image_builder/image_builder/verify.py

```diff
@@ -12,11 +12,12 @@
 logger = get_logger(__name__)
 
 
-def verify_manifest_consistency(submitted_yaml: str, workspace: Path, repo_dir: Path):
+def verify_manifest_consistency(submitted_yaml: str, workspace: Path, repo_dir: Path) -> str | None:
     repo_file = _find_workspace_manifest_file(workspace, repo_dir)
     if not repo_file:
-        logger.warning("Can't find manifest file in a Job repository to verify it with submitted YAML")
-        return
+        warning = "Can't find manifest file in a Job repository to verify it with submitted YAML"
+        logger.warning(warning)
+        return warning
```

Comment:
Can't add comment to unchanged lines, so I'll add it here:
Since this function returns warnings instead of raising errors, I think this:
```python
raise RuntimeError('Submitted job manifest is not consistent with the file found in a repository. '
                           'Did you forget to do "git push"? '
                           f'Difference: {difference}')
```
 is now supposed to be
```python
return 'Submitted job manifest is not consistent with the file found in a repository. '
                           'Did you forget to do "git push"? '
                           f'Difference: {difference}'
```

Reply:
Earlier the function logged a warning, so I assumed we wanted the warning. I agree that the above should be a warning instead of an error.